src/vcProject.cpp

```diff
@@ -129,6 +133,50 @@ void vcProject_Deinit(vcState *pProgramData, vcProject *pProject)
   vdkProject_Release(&pProject->pProject);
 }
 
+void vcProject_Save(vcState *pProgramState, const char *pPath, bool allowOverride)
+{
+  if (pProgramState == nullptr || pPath == nullptr)
+    return;
+
+  const char *pOutput = nullptr;
+
+  if (pProgramState->gis.isProjected)
+    vdkProjectNode_SetMetadataInt(pProgramState->activeProject.pRoot, "defaultcrs", pProgramState->gis.SRID);
+
+  if (vdkProject_WriteToMemory(pProgramState->activeProject.pProject, &pOutput) == vE_Success)
+  {
+    udFindDir *pDir = nullptr;
+    udFilename exportFilename(pPath);
+
+    if (!udStrEquali(pPath, "") && !udStrEndsWithi(pPath, "/") && !udStrEndsWithi(pPath, "\\") && udOpenDir(&pDir, pPath) == udR_Success)
+      exportFilename.SetFromFullPath("%s/untitled_project.json", pPath);
+    else if (exportFilename.HasFilename())
+      exportFilename.SetExtension(".json");
+    else
+      exportFilename.SetFilenameWithExt("untitled_project.json");
+
+    // Check if file path exists before writing to disk, and if so, the user will be presented with the option to overwrite or cancel
+    if (allowOverride || vcModals_OverwriteExistingFile(exportFilename.GetPath()))
+    {
+      vcState::ErrorItem projectError;
```

Comment:
Potential uninitialized memory

Reply:
This is currently just moved code- do you want me to also make fixes (which complicates this MR)?